util/chplenv/chpl_atomics.py

```diff
@@ -38,6 +38,8 @@ def get(flag='target'):
                     atomics_val = 'intrinsics'
                 elif version >= CompVersion('4.1') and not platform_val.endswith('32'):
                     atomics_val = 'intrinsics'
+            elif compiler_val == 'aarch64-gnu':
+                atomics_val = 'cstdlib'
```

Comment:
For the future, this could be based on the compiler version and default to locks, or intrinsics could be modified.